### PR TITLE
Actually fixed dockerng.list_tags

### PR DESCRIPTION
**Pull request by farcaller**
### What does this PR do?

Fixes the bug introduced in #34702

### CODE REVIEW

File: salt/modules/dockerng.py [hunk 1]
```diff
@@ -2237,7 +2237,9 @@ def list_tags():
     '''
     ret = set()
     for item in six.itervalues(images()):
-        for repo_tag in item.get('RepoTags', []):
+        if item.get('RepoTags', None):
```
Comment (clinta) [inline salt/modules/dockerng.py]: I think you're missing a `not` here.

Comment (farcaller) [inline salt/modules/dockerng.py]: Indeed. I should have actually tested that version. Or this version, FWIW.

File: salt/modules/dockerng.py [hunk 2]
```diff
@@ -2237,7 +2237,9 @@ def list_tags():
     '''
     ret = set()
     for item in six.itervalues(images()):
-        for repo_tag in item.get('RepoTags', []):
+        if not item.get('RepoTags', None):
```
Comment (hjc1710) [inline salt/modules/dockerng.py]: `None` is the default value that `dict.get` uses, no need to repeat it.

Comment (farcaller) [inline salt/modules/dockerng.py]: Ack.

**Pull request merged**